src/installer/corehost/corehost.cpp

```diff
@@ -212,12 +212,24 @@ int exe_start(const int argc, const pal::char_t* argv[])
         trace::info(_X("Dotnet path: [%s]"), dotnet_root.c_str());
         trace::info(_X("App path: [%s]"), app_path.c_str());
 
-        hostfxr_set_error_writer_fn set_error_writer_fn = reinterpret_cast<hostfxr_set_error_writer_fn>(pal::get_symbol(fxr, "hostfxr_set_error_writer"));
-
         {
-            propagate_error_writer_t propagate_error_writer_to_hostfxr(set_error_writer_fn);
+            auto custom_error_writer = pal::get_symbol(fxr, "hostfxr_set_error_writer");
+
+            if (custom_error_writer)
+            {
+                hostfxr_set_error_writer_fn set_error_writer_fn = reinterpret_cast<hostfxr_set_error_writer_fn>(custom_error_writer);
+                propagate_error_writer_t propagate_error_writer_to_hostfxr(set_error_writer_fn);
+            }
 
             rc = main_fn_v2(argc, argv, host_path_cstr, dotnet_root_cstr, app_path_cstr);
+
+            if (rc == StatusCode::FrameworkMissingFailure && !custom_error_writer)
+            {
+                pal::string_t url = get_download_url(nullptr, nullptr);
```

Comment:
I don't understand the need for the special prefixes and different new-line handling. I would expect the behavior to be identical regardless of whether we got the error through the custom error writer or directly from here.

We could mimic the error produced by https://github.com/dotnet/runtime/blob/66cb5f9a1bdbb0fd0b357a0dbc862daa0b45e85a/src/installer/corehost/cli/fxr/fx_resolver.messages.cpp#L93. The framework name pretty much has to be `Microsoft.NETCore.App` and the version can be something like the current version of the apphost (so `COMMON_HOST_PKG_VER` for example - as per https://github.com/dotnet/runtime/blob/66cb5f9a1bdbb0fd0b357a0dbc862daa0b45e85a/src/installer/corehost/cli/apphost/apphost.windows.cpp#L98). Maybe we could even share that code to produce the same error message (but it would probably require some tweaking).